### PR TITLE
Dropdown component

### PR DESCRIPTION
**Pull request by Saionaro**

### CODE REVIEW

File: components/Dropdown/Dropdown.tsx [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    },
+  }),
+  all,
+  variant({
+    variants: {
+      dark: {
+        color: "black",
+        borderColor: ["gray", "gray"],
+        "&:focus-within, &:focus, &:hover": {
+          backgroundColor: "light-purple",
```
Comment (iAdramelk) [inline components/Dropdown/Dropdown.tsx:97]:
```suggestion
          bg: "light-purple",
```

File: components/Dropdown/Dropdown.tsx [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+    "&:focus-within, &:hover, &:active, &:focus": {
+      borderColor: "white",
+    },
+  }),
+  all,
+  variant({
+    variants: {
+      dark: {
+        color: "black",
+        borderColor: ["gray", "gray"],
```
Comment (iAdramelk) [inline components/Dropdown/Dropdown.tsx:95]:
```suggestion
        borderColor: "gray",
```

File: components/Dropdown/Dropdown.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  component: Component,
+  title: "Dropdown",
+};
+
+export const DarkDropdown = StoryComponent.bind({});
+
+DarkDropdown.args = {
+  options: ["Hey", "How", "Are", "You"],
+  variant: "dark",
+  onChange: console.log,
```
Comment (iAdramelk) [inline components/Dropdown/Dropdown.stories.tsx:33]: Please use actions https://storybook.js.org/docs/react/essentials/actions#action-args

File: components/Dropdown/Dropdown.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    {
+      id: "4",
+      title: "Title 4",
+    },
+  ],
+  variant: "dark",
+  pickValue: (item) => item.id,
+  renderOption: (item) => item.title,
+  onChange: console.log,
+};
```
Comment (iAdramelk) [inline components/Dropdown/Dropdown.stories.tsx:69]: Please add example with `icon` field.

**Review by iAdramelk — CHANGES_REQUESTED** [review]

File: components/Dropdown/Dropdown.tsx (hunk 2 above)
Comment (Saionaro) [reply components/Dropdown/Dropdown.tsx:95]: Unfortunately we want it to override this line:
https://github.com/gravitational/next/blob/6f00a0ddf902e4f62ea2fdd7bd0e8c30d4163777/components/Dropdown/Dropdown.tsx#L77

**Pull request merged**